src/ejabberd_c2s.erl

```diff
@@ -435,8 +435,29 @@ filter_mechanism(<<"EXTERNAL">>, S) ->
         error -> false;
         _ -> true
     end;
+filter_mechanism(<<"SCRAM-SHA-1-PLUS">>, S) ->
+    is_channel_binding_supported(S);
+filter_mechanism(<<"SCRAM-SHA-", _N:3/binary, "-PLUS">>, S) ->
+    is_channel_binding_supported(S);
 filter_mechanism(_, _) -> true.
 
+is_channel_binding_supported(State) ->
+    TLSEnabled = State#state.tls_enabled,
+    TLSRequired = State#state.tls_required,
+    Socket = State#state.socket,
+    case TLSEnabled or not TLSRequired of
+        true ->
+            SockMod = (State#state.sockmod):get_sockmod(Socket),
+            is_fast_tls_configured(SockMod, Socket);
+        false ->
+            false
+    end.
```

Comment:
Do we need this case at all? Can we assume that if the `SockMod` is `ejabberd_tls` the encryption is already enabled and we can check if the backend module is `fast_tls`? If the `SockMod` is not `ejabberd_tls`, this means encryption is not enabled, so channel binding is not possible at this point.
What am I missing?
```suggestion
    SockMod = (State#state.sockmod):get_sockmod(Socket),
    is_fast_tls_configured(SockMod, Socket);
```

Reply:
Yep, you're right, I was trying to be too cautious. I changed as suggested and works for login SUITE. Will see on CI if still works for other SUITEs.